js/stringProcessing/indices.js

```diff
@@ -15,12 +16,19 @@ function getIndicesByWord( word, text ) {
 	var searchStringLength = word.length;
 	var index, indices = [];
 	while ( ( index = text.indexOf( word, startIndex ) ) > -1 ) {
-		indices.push(
-			{
-				index: index,
-				match: word,
-			}
-		);
+		// Check if the previous and next character are word boundaries to determine if a complete word was detected
+		var previousCharacter = characterInBoundary( text[ index - 1 ] ) || index === 0;
```

Comment:
`isPreviousCharacterBoundary` and `isNextCharacterBoundary` seem better variable names.